libs/block-content/api/src/lib/teaser/slot-teasers-loader.ts

```
@@ -0,0 +1,147 @@
+    return []
+  }
+
+  addLoadedTeaser(...teaser: (typeof Teaser)[]) {
+    this.loadedTeasers.push(...teaser)
+  }
+
+  getLoadedTeasers(type: TeaserType): string[] {
+    const ids = this.loadedTeasers.reduce((ids: string[], teaser) => {
+      if (teaser.type === type) {
+        if (teaser.type === TeaserType.Article && teaser.articleID) {
+          return [...ids, teaser.articleID]
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)